File: weather-mvp-dagger/src/main/java/com/baronzhang/android/weather/feature/contract/SelectCityContract.java
package com.baronzhang.android.weather.feature.contract;

import java.util.List;

import com.baronzhang.android.weather.model.db.entities.City;

/**
 * @author baronzhang (baron[dot]zhanglei[at]gmail[dot]com)
 */
public interface SelectCityContract {

    interface View extends BaseView<Presenter> {

        void displayCities(List<City> cities);
    }

    interface Presenter extends BasePresenter {

        void loadCities();
    }
}

File: weather-mvp-dagger-kotlin/src/main/java/com/minimalist/weather/kotlin/feature/contract/SelectCityContract.kt
package com.minimalist.weather.kotlin.feature.contract


import com.minimalist.weather.kotlin.model.data.entity.city.City

interface SelectCityContract {

    interface View : BaseView {

        fun displayCities(cities: List<City>)
    }

    interface Presenter : BasePresenter<View> {

        fun loadCities()
    }
}
